addons/controllers/antreaconfig_controller_test.go

```diff
@@ -227,4 +254,212 @@ var _ = Describe("AntreaConfig Reconciler and Webhooks", func() {
+			})
+
+		})
+
+	})
+
+	When("custom clusterbootstrap points to precreated antrea config", func() {
+		BeforeEach(func() {
+			clusterName = "custom-cb-2-cluster"
+			clusterNamespace = "custom-cb-2-namespace"
+			configName = "antrea-config-custom-2"
+			clusterBoostrapFile = "testdata/antrea-custom-cb.yaml"
+			clusterInfraName = "custom-cb-docker-cluster-2"
+			tkrString = "v1.23.2"
+			clusterResourceFilePath = "testdata/antrea-custom-cb-2-resources.yaml"
+			newDefaultMTU = "8902"
+			ns := &v1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: clusterNamespace,
+				},
+			}
+			err := k8sClient.Create(ctx, ns)
+			if err != nil {
+				Expect(apierrors.IsAlreadyExists(err)).To(BeTrue())
+			}
+
+		})
+		It("should add cluster as owner reference to precreated antrea config", func() {
+			By("lets make sure resources are available", func() {
+				kappConfig := &runtanzuv1alpha3.KappControllerConfig{}
+				kappConfigKey := client.ObjectKey{Name: "test-cluster-custom-cb-2-kapp-controller-config",
+					Namespace: addonNamespace}
+				err := k8sClient.Get(ctx, kappConfigKey, kappConfig)
+				Expect(err).ToNot(HaveOccurred())
+
+			})
+
+			By("Create antrea config in the cluster's namespace with random name", func() {
+				datavalues := &cniv1alpha1.AntreaConfigDataValue{DefaultMTU: newDefaultMTU}
+				antreaConfig := genearateAntreaConfig(configName, clusterNamespace, datavalues)
+				err := k8sClient.Create(ctx, antreaConfig)
+				Expect(err).ToNot(HaveOccurred())
+				err = k8sClient.Get(ctx, client.ObjectKeyFromObject(antreaConfig), antreaConfig)
+				Expect(err).ToNot(HaveOccurred())
+			})
+
+			By("Create custom clusterbootstrap in clusters namespace", func() {
+				f, err := os.Open(clusterBoostrapFile)
+				Expect(err).ToNot(HaveOccurred())
+				defer f.Close()
+				Expect(testutil.CreateResources(f, cfg, dynamicClient)).To(Succeed())
+
+			})
+
+			By("Create the cluster", func() {
+				cluster := generateDockerCluster(clusterName, clusterNamespace, tkrString, clusterInfraName)
+				err := k8sClient.Create(ctx, cluster)
+				Expect(err).ToNot(HaveOccurred())
+				err = k8sClient.Get(ctx, client.ObjectKeyFromObject(cluster), cluster)
+				Expect(err).ToNot(HaveOccurred())
+
+			})
+
+			By("Verify contents of resulting antrea config", func() {
+				// eventually the secret ref to the data values should be updated
+				antreaConfig := &cniv1alpha1.AntreaConfig{}
+				Eventually(func() error {
+					configKey := client.ObjectKey{
+						Namespace: clusterNamespace,
+						Name:      configName,
+					}
+					if err := k8sClient.Get(ctx, configKey, antreaConfig); err != nil {
+						return fmt.Errorf("failed to get antrea config '%v': '%v'", configKey, err)
+					}
+					if antreaConfig.Status.SecretRef == "" {
+						return fmt.Errorf("antrea config status not yet updated: %v", configKey)
+					}
+					Expect(antreaConfig.Status.SecretRef).To(Equal(fmt.Sprintf("%s-%s-data-values", clusterName, constants.AntreaAddonName)))
```

Comment:
There should be no Expect inside an Eventually node. It shortcircuits the evenually.
also change the way the clusters  data value secret name is deduced.  It should use the util.Generate.... fucntions